darwin/Platform.c

```diff
@@ -63,6 +62,12 @@ in the source distribution for its full text.
 #include <mach/mach_time.h>
 #endif
 
+#ifdef HAVE_SYS__TYPES__MACH_PORT_T_H
+#include <sys/_types/_mach_port_t.h>
```

Comment:
I'm surprised by the inclusion of a system header whose name begins with an underscore. I always assumed that meant it was a private header, not intended to be included directly and subject to change in the future.

Reply:
@ryandesign This is not my addition though. It was added in https://github.com/htop-dev/htop/commit/ed7eac5dfe8239f7942102f9d1a4f9ef5210f4d7 and has broken everything which does not have this header. I am merely trying to fix it for those systems without changing the code for the new ones.

Reply:
@barracuda156 Exactly, you correctly noted that including `<sys/_types/_mach_port_t.h>` causes a problem, but your proposal makes the code include that header sometimes and a different header at other times. My question is whether it wouldn't be better never to include `<sys/_types/_mach_port_t.h>` and to instead always include whatever header is documented to provide whatever the functionality is that is needed here.

I assume that the wanted functionality is the definition of `mach_port_t` since that's the only thing in `<sys/_types/_mach_port_t.h>` in the macOS 11 SDK. I've never found the Mach stuff to be documented very well. [Apple's site documents](https://developer.apple.com/documentation/kernel/mach_port_t) that `mach_port_t` has existed since Mac OS X 10.0 but nothing else about it, such as what header to include. Maybe it used to be better documented and Apple removed that documentation at some point.

grepping through headers, I see that `_mach_port_t.h` is included by `<pthread/pthread.h>` and `<mach/port.h>`, so I'm guessing the latter is the one that should always be included.

Reply:
@ryandesign Could you confirm it works for Intel? If yes, then we should just go with that, and drop all fancy checks.

Reply:
I can't test it at the moment but it seems like it should work.

Oh, and since `<mach/port.h>` is documented to exist since the beginning of time you won't need to check for its existence.

Reply:
It looks like htop has a GitHub Actions workflow set up that builds on macOS, so hopefully that is sufficient to verify any proposed change does not break the build on current macOS.

Reply:
Ok let me rebase.